src/compiler/resolver_method.cc

```diff
@@ -2968,31 +2969,39 @@ void MethodResolver::_visit_potential_call(ast::Node* ast_target,
     }
   }
 
-  if ((ast_target->is_Identifier() && !is_literal_super(ast_target)) ||
-      _scope->is_prefixed_identifier(ast_target) ||
-      _scope->is_static_identifier(ast_target)) {
-    _visit_potential_call_identifier(ast_target,
-                                     call_builder,
-                                     named_lsp_selection,
-                                     target_name_node,
-                                     target_name);
-  } else if (ast_target->is_Dot() && !is_constructor_super_call) {
-    _visit_potential_call_dot(ast_target->as_Dot(),
-                              call_builder,
-                              named_lsp_selection);
-  } else if (ast_target->is_Index()) {
-    _visit_potential_call_index(ast_target->as_Index(), call_builder);
-  } else if (ast_target->is_IndexSlice()) {
-    _visit_potential_call_index_slice(ast_target->as_IndexSlice(), call_builder);
-  } else if (is_literal_super(ast_target) ||
-             (ast_target->is_Dot() && is_constructor_super_call)) {
-    _visit_potential_call_super(ast_target, call_builder, is_constructor_super_call);
+  if (potential_call->is_Index()) {
```

Comment:
This PR is significantly easier to review with whitespace changes turned off.
I moved a chunk into an 'else' branch and thus nested it.